### PR TITLE
ENG-2767 react hook form submission

### PR DESCRIPTION
**Pull request by agiron123**
## Describe your changes and why you are making these changes
Putting up a quick draft to review my work and collect thoughts on how to best refactor and document a reference implementation.

## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [x] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [x] If this is a new feature, I have added unit tests and integration tests.
- [x] I have run the integration tests locally and they are passing.
- [x] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [x] All features on the UI continue to work correctly.
- [x] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/ui/common/src/components/integrations/connectedIntegrations.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/connectedIntegrations.tsx:49]: Thoughts on architecture:
-Having Integration be the actual integration component and IntegrationCard serve more as a container component seems like a good way to architect this.

File: src/ui/common/src/components/integrations/dialogs/IntegrationFileUploadField.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/dialogs/IntegrationFileUploadField.tsx:42]: This is one approach that we can use here. Might also consider the approach used in NotificationsLevel component. It's not as clever I'd say, but I think the pattern there is simple and more in line with a KISS approach.

File: src/ui/common/src/components/integrations/dialogs/IntegrationTextInputField.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/dialogs/IntegrationTextInputField.tsx:41]: NOTE: When an error comes back in validation in react-hook-form, it will be a key of the errors object.

e.g. 
errors.foo = "some error object".
TODO: add errors[key] check to IntegrationtextInputField to render errors appropriately.

File: src/ui/common/src/components/integrations/dialogs/airflowDialog.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/dialogs/airflowDialog.tsx:39]: TODO: Use register and setValue here to get the logic for removing the http:// or https:// prefix from the url.

File: src/ui/common/src/components/integrations/addIntegrations.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/addIntegrations.tsx:33]: remove comment

File: src/ui/common/src/components/integrations/cards/card.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/cards/card.tsx:36]: Shall we remove? May be a question for the larger team here.

I think we should remove if this is no longer used.

File: src/ui/common/src/components/integrations/cards/card.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/cards/card.tsx:97]: File a Linear task here and we can resolve later, as @kenxu95 is working on the cards at the moment.

File: src/ui/common/src/components/integrations/connectedIntegrations.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/connectedIntegrations.tsx:49]: remove log.

File: src/ui/common/src/components/integrations/dialogs/IntegrationTextInputField.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/dialogs/IntegrationTextInputField.tsx:39]: remove log and comments.

File: src/ui/common/src/components/integrations/dialogs/airflowDialog.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/dialogs/airflowDialog.tsx:20]: remove comments.

File: src/ui/common/src/components/integrations/dialogs/airflowDialog.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/dialogs/airflowDialog.tsx:29]: remove comment

File: src/ui/common/src/components/integrations/dialogs/athenaDialog.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/dialogs/athenaDialog.tsx:54]: remove log and comments.

File: src/ui/common/src/components/integrations/dialogs/athenaDialog.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/dialogs/athenaDialog.tsx:36]: remove comments.

File: src/ui/common/src/components/integrations/dialogs/athenaDialog.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/dialogs/athenaDialog.tsx:8]: remove comment

File: src/ui/common/src/components/integrations/dialogs/athenaDialog.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/dialogs/athenaDialog.tsx:64]: Debugging notes on fixing the file upload errors:
TODO: log config_file_conttent. see what we're getting back here after upload.

File: src/ui/common/src/components/integrations/dialogs/athenaDialog.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/dialogs/athenaDialog.tsx:169]: remove comment

File: src/ui/common/src/components/integrations/cards/card.tsx
Comment (agiron123) [reply src/ui/common/src/components/integrations/cards/card.tsx:97]: See ENG-2768

File: src/ui/common/src/components/integrations/dialogs/athenaDialog.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/dialogs/athenaDialog.tsx:251]: remove todo.

File: src/ui/common/src/components/integrations/dialogs/awsDialog.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/dialogs/awsDialog.tsx:36]: remove comment.

File: src/ui/common/src/components/integrations/dialogs/awsDialog.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/dialogs/awsDialog.tsx:49]: remove comments.

File: src/ui/common/src/components/integrations/dialogs/bigqueryDialog.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/dialogs/bigqueryDialog.tsx:19]: remove comments.

File: src/ui/common/src/components/integrations/dialogs/bigqueryDialog.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/dialogs/bigqueryDialog.tsx:31]: Finish debugging file input here and then remove comments if no longer needed.

File: src/ui/common/src/components/integrations/dialogs/databricksDialog.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/dialogs/databricksDialog.tsx:21]: remove comments.

File: src/ui/common/src/components/integrations/dialogs/dialog.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/dialogs/dialog.tsx:182]: TODO: Get forms to actually submit as they were before.

File: src/ui/common/src/components/integrations/dialogs/dialog.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/dialogs/dialog.tsx:231]: clean this up / remove.

Figure out what's needed for the GCS Dialog first since it had some special logic there.

File: src/ui/common/src/components/integrations/dialogs/dialog.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/dialogs/dialog.tsx:449]: remove comment

File: src/ui/common/src/components/integrations/dialogs/dialog.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/dialogs/dialog.tsx:459]: remove comment

File: src/ui/common/src/components/integrations/dialogs/dialog.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/dialogs/dialog.tsx:495]: remove comment

File: src/ui/common/src/components/integrations/dialogs/dialog.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/dialogs/dialog.tsx:526]: remove logs and get form submission to work properly :)

File: src/ui/common/src/components/integrations/dialogs/emailDialog.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/dialogs/emailDialog.tsx:47]: TODO: Get this fixed up.

File: src/ui/common/src/components/integrations/dialogs/gcsDialog.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/dialogs/gcsDialog.tsx:116]: remove comments and clean up below.

File: src/ui/common/src/components/integrations/dialogs/kubernetesDialog.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/dialogs/kubernetesDialog.tsx:39]: TODO: Add link to linear task for this

File: src/ui/common/src/components/integrations/dialogs/lambdaDialog.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/dialogs/lambdaDialog.tsx:16]: remove comments.

File: src/ui/common/src/components/integrations/dialogs/lambdaDialog.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/dialogs/lambdaDialog.tsx:40]: Add validation function here. role_arn is a required field that we should be checking for.

File: src/ui/common/src/components/integrations/dialogs/mariadbDialog.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/dialogs/mariadbDialog.tsx:5]: remove comment.

File: src/ui/common/src/components/integrations/dialogs/mariadbDialog.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/dialogs/mariadbDialog.tsx:21]: remove comments

File: src/ui/common/src/components/integrations/dialogs/mongoDbDialog.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/dialogs/mongoDbDialog.tsx:17]: remove comments

File: src/ui/common/src/components/integrations/dialogs/mysqlDialog.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/dialogs/mysqlDialog.tsx:20]: remove comments

File: src/ui/common/src/components/integrations/dialogs/redshiftDialog.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/dialogs/redshiftDialog.tsx:20]: remove comments

File: src/ui/common/src/components/integrations/dialogs/s3Dialog.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/dialogs/s3Dialog.tsx:56]: remove comment

File: src/ui/common/src/components/integrations/dialogs/s3Dialog.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/dialogs/s3Dialog.tsx:68]: make sure file upload is working properly here and remove comments when done.

File: src/ui/common/src/components/integrations/dialogs/slackDialog.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/dialogs/slackDialog.tsx:33]: remove comments.

File: src/ui/common/src/components/integrations/dialogs/slackDialog.tsx
Comment (agiron123) [inline src/ui/common/src/components/integrations/dialogs/slackDialog.tsx:139]: remove comment

File: src/ui/common/src/components/integrations/dialogs/airflowDialog.tsx
Comment (agiron123) [reply src/ui/common/src/components/integrations/dialogs/airflowDialog.tsx]: resolved.

File: src/ui/common/vite.config.ts
Comment (likawind) [inline src/ui/common/vite.config.ts:19]: What's the context of this change?

File: sdk/aqueduct/models/operators.py
Comment (likawind) [inline sdk/aqueduct/models/operators.py:214]: This is a `.py` change? Is it intended?

**Review by likawind — COMMENTED** [review]
@agiron123 codes makes sense overall! I would approve once I do a more closer pass on details. Before we merge, I'm wondering how did you test all changes? Did you manually go over all dialogs? If you've tested things to a level where you are confident, another idea is to merge as-is and use the demo hour to have ppl play around the new dialogs and file issues accordingly.

File: src/ui/common/src/components/integrations/dialogs/emailDialog.tsx
Comment (likawind) [inline src/ui/common/src/components/integrations/dialogs/emailDialog.tsx:169]: remove these

**Review by likawind — APPROVED** [review]
:shipit:

**Pull request merged**
